src/regex/match_data.cr

```diff
@@ -61,11 +61,11 @@ class Regex
     # subpattern is unused.
     #
     # ```
-    # "Crystal".match(/r/).not_nil!.begin(0)     # => 1
-    # "Crystal".match(/r(ys)/).not_nil!.begin(1) # => 2
-    # "クリスタル".match(/リ(ス)/).not_nil!.begin(0)    # => 1
-    # "Crystal".match(/r/).not_nil!.begin(1)     # IndexError: Invalid capture group index: 1
-    # "Crystal".match(/r(x)?/).not_nil!.begin(1) # IndexError: Capture group 1 was not matched
+    # "Crystal".match!(/r/).begin(0)        # => 1
+    # "Crystal".match!(/r(ys)/).begin(1)    # => 2
+    # "クリスタル".match!(/リ(ス)/).begin(0)  # => 1
+    # "Crystal".match!(/r/).begin(1)     # IndexError: Invalid capture group index: 1
+    # "Crystal".match!(/r(x)?/).begin(1) # IndexError: Capture group 1 was not matched
```

Comment:
This was already the case before this PR, but those return comments are misaligned